PopulateResponseInfo.pas

```diff
@@ -0,0 +1,111 @@
+unit PopulateResponseInfo;
+
+interface
+
+uses
+  Data.Cloud.CloudAPI,
+  IPPeerAPI;
+
+type
+  TPopulateResponseInfo = class
+  public
+    procedure FromPeer(const ResponseInfo: TCloudResponseInfo; const Peer: IIPHTTP); overload;
+    procedure FromExceptionPeer(const ResponseInfo: TCloudResponseInfo; const ExceptionPeer: EIPHTTPProtocolExceptionPeer); overload;
+  end;
+
+implementation
+
+uses
+  Classes,
+  SysUtils,
+  XmlIntf,
+  XmlDoc;
+
+{ TPopulateResponseInfo }
+
+procedure TPopulateResponseInfo.FromExceptionPeer(const ResponseInfo: TCloudResponseInfo;
+  const ExceptionPeer: EIPHTTPProtocolExceptionPeer);
+const
+  NODE_ERRORS = 'Errors';
+  NODE_ERROR = 'Error';
+  NODE_ERROR_MESSAGE = 'Message';
+  NODE_ERROR_CODE = 'Code';
+  NODE_REQUEST_ID = 'RequestId';
+  NODE_RESPONSE_METADATA = 'ResponseMetadata';
+var
+  XmlDoc: IXMLDocument;
+  IsErrors: Boolean;
+  Aux, ErrorNode, MessageNode: IXMLNode;
+  ErrorCode, ErrorMsg: string;
+begin
+  if ExceptionPeer.ErrorMessage = EmptyStr then
```

Comment:
`EmptyStr` this just makes it harder to read, I would use `= ''` or 'ErrorMessage.IsEmpty`